### PR TITLE
feat: Password based Authentication

### PR DESCRIPTION
**Pull request by BhanuPratap-Singh-Naruka**
### Task Link

https://www.notion.so/BE-Password-Based-Authentication-feature-6cc5095dc33f40e38eb9d316c8d91d70?pvs=4

### Description
* Added Password input field to login and signup UI  
* Changes in Login and signup API routes for password validation  
* Some code refactoring

### Screenshot / Gif (If Any)

password  

* If we enter one of the input wrong: -  
  
password1

### TODO Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

**Comment by abhimishra01** [comment]
Plz Change the PR title to :- `feat: password based authentication`

File: .eslintrc.js [hunk 1]
```diff
@@ -21,5 +21,7 @@ module.exports = {
   plugins: ['react', '@typescript-eslint', 'prettier'],
   rules: {
     'react/react-in-jsx-scope': 0,
+    'import/prefer-default-export': 0,
+    '@typescript-eslint/no-non-null-assertion': 0,
```
Comment (abhimishra01) [inline .eslintrc.js:25]: What does this do?

File: .eslintrc.js [hunk 2]
```diff
@@ -21,5 +21,7 @@ module.exports = {
   plugins: ['react', '@typescript-eslint', 'prettier'],
   rules: {
     'react/react-in-jsx-scope': 0,
+    'import/prefer-default-export': 0,
```
Comment (abhimishra01) [inline .eslintrc.js:24]: Nice

**Review by abhimishra01 — APPROVED** [review]
Lgtm

File: .eslintrc.js (hunk 1 above)
Comment (BhanuPratap-Singh-Naruka) [reply .eslintrc.js:25]: In Users.ts we are passing this:-  
```
const streamChat = StreamChat.getInstance(
  process.env.STREAM_API_KEY!, 
  process.env.STREAM_PRIVATE_API_KEY
);
```
process.env.STREAM_API_KEY and we are putting ! behind this to tell the server that this value is defined somewhere and it won't be null. But in esLint it is forbidden to use (!) for this reason. And it was giving error so I fixed it by using 
```
'@typescript-eslint/no-non-null-assertion': 0,
```

File: .eslintrc.js (hunk 2 above)
Comment (BhanuPratap-Singh-Naruka) [reply .eslintrc.js:24]: Thank you.

File: .eslintrc.js (hunk 1 above)
Comment (abhimishra01) [reply .eslintrc.js:25]: Ok

**Pull request merged**